lib/util/transport.js

```diff
@@ -158,6 +197,27 @@ var transport_decode = function (buffer) {
 
     while (off < buffer.length) {
 
+        if (buffer.slice(0, 10).toString() === "PACKED_XML") {
+            try {
+                ct = packed_xml.unpack(buffer);
+                ct = decode_packed_xml_contents(ct);
+            } catch (e) {
+                // let the user know that something wrong,
+                // as exception will be eaten later...
+                console.error(e.message);
+                throw e;
+            }
+            ret.header = {};
+            for (var doc in ct) {
```

Comment:
Some 3rd-party libraries can monkey patch `Object.prototype` or `Array.prototype` to add some methods and they would appear as keys in this loop.

Consider this example:

```
var x = {a: 1, b: 2, c: 3}
for (name in x) {
  console.log(name)
}

Output:
a
b
c
```

And now some cool guy changes prototype:

```
>Object.prototype.r = 4
>for (name in x) {
  console.log(name)
}

Output:
a
b
c
r // BOOM!
```

On the opposite, when using Object.keys:

```
> Object.keys(x)
["a", "b", "c"]
```